File: app/src/main/java/net/squanchy/eventdetails/EventDetailsModule.java
package net.squanchy.eventdetails;

import net.squanchy.service.firebase.FirebaseAuthService;
import net.squanchy.service.repository.EventRepository;

import dagger.Module;
import dagger.Provides;

@Module
class EventDetailsModule {
    
    @Provides
    EventDetailsService scheduleService(EventRepository eventRepository, FirebaseAuthService authService) {
        return new EventDetailsService(eventRepository, authService);
    }
}

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsModule.kt
package net.squanchy.eventdetails

import dagger.Module
import dagger.Provides
import net.squanchy.service.repository.AuthService
import net.squanchy.service.repository.EventRepository

@Module
internal class EventDetailsModule {

    @Provides
    fun eventDetailsService(eventRepository: EventRepository, authService: AuthService): EventDetailsService {
        return EventDetailsService(eventRepository, authService)
    }
}
